fhir-server/src/main/java/au/csiro/pathling/spark/Spark.java

```diff
@@ -43,14 +45,20 @@ public class Spark {
    * creation
    * @param environment Spring {@link Environment} from which to harvest Spark configuration
    * @param sparkListener a {@link SparkListener} that is used to monitor progress of jobs
+   * @param sqlFunction1 a list of {@link SqlFunction1} that should be registered
+   * @param sqlFunction2 a list of {@link SqlFunction2} that should be registered
+   * @param sqlFunction3 a list of {@link SqlFunction3} that should be registered
    * @return A shiny new {@link SparkSession}
    */
   @Bean(destroyMethod = "stop")
   @ConditionalOnMissingBean
   @Nonnull
   public static SparkSession build(@Nonnull final Configuration configuration,
       @Nonnull final Environment environment,
-      @Nonnull final Optional<SparkListener> sparkListener) {
+      @Nonnull final Optional<SparkListener> sparkListener,
+      @Nonnull final List<SqlFunction1> sqlFunction1,
+      @Nonnull final List<SqlFunction2> sqlFunction2,
+      @Nonnull final List<SqlFunction3> sqlFunction3) {
     log.debug("Creating Spark session");
```

Comment:
Does Spring automatically create these lists from all the compatible components?

Reply:
Yes it does.